Framework/Built_In_Automation/Performance_Testing/BuiltInFunctions.py

```diff
@@ -0,0 +1,225 @@
+            if None in [losust_var_name,taskset_name]: 
+                CommonUtil.ExecLog(sModuleInfo,  f"dataset is inaccurate", 3)
+                return "zeuz_failed"
+            locust_var = sr.Get_Shared_Variables(losust_var_name,log=False)
+            locust_var['task_sets'][taskset_name] = {'sequential':sequential,'tasks':[]}
+            for user_name in user_names:
+                locust_var['users'][user_name]['user_task_sets'].append(taskset_name)
+        except:
+            CommonUtil.ExecLog(sModuleInfo, "Failed to parse data.", 1)
+            traceback.print_exc()
+            return "zeuz_failed"
+        sr.Set_Shared_Variables(losust_var_name, locust_var)
+        return "passed"
+    except:
+        return CommonUtil.Exception_Handler(sys.exc_info())
+
+
+@logger
+def assign_locust_task(data_set):
+    """
+    Add task configuration to the existing locust user
+    This action provides that data to define the tasks performed by locust user/class
+    Args:
+        data_set:
+          action             | input parameter    | get/post etc
+          data               | input parameter    | url of endpoint
+          task name          | input parameter    | string
+          name               | input parameter    | existing user/class name Case sensitive
+          assign locust task | performance action | existing locust variable name
+    Returns:
+        "passed" if success.
+        "zeuz_failed" otherwise.
+    """
+
+    sModuleInfo = inspect.currentframe().f_code.co_name + " : " + MODULE_NAME
+
+    try:
+        locust_var = locust_var_name = action = url = user_name = task_name = taskset_name = weight = None
+        try:
+            for left, mid, right in data_set:
+                left = left.strip().lower()
+                if mid.strip().lower() == "input parameter":
+                    if "action" == left.strip():
+                        action = right.strip().lower()
+                    elif "url" == left.strip():
+                        url = right.strip().lower()
+                    elif "user name" == left.strip():
+                        user_name = right.strip()
+                    elif "taskset name" == left.strip():
+                        taskset_name = right.strip()
+                    elif "task name" == left:
+                        task_name = right.strip().lower()
+                if mid.strip().lower() == "parameter":
+                    if "weight" == left.strip():
+                        weight = right.strip().lower()
+                elif mid.strip().lower() == "action":
+                    if "assign locust task" == left:
+                        locust_var_name = right.strip()
+            if None in [locust_var_name,action,url,task_name]:
+                CommonUtil.ExecLog(sModuleInfo,  f"dataset is inaccurate", 3)
+                return "zeuz_failed"
+            if (sum(x is not None for x in [user_name,taskset_name])) != 1:
```

Comment:
extra brackets not needed. sometimes in other cases it may convert the datatype to tuple

Reply:
@Oytizzo